x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/index.tsx

```diff
@@ -45,34 +45,47 @@ const TIMELINE_MARGINS = {
 
 interface Props {
   agentMarks: AgentMark[];
-  urlParams: IUrlParams;
+  waterfallItemId?: string;
   waterfall: IWaterfall;
   location: Location;
-  serviceColors: IServiceColors;
   exceedsMax: boolean;
 }
 
-export class Waterfall extends Component<Props> {
-  public onOpenFlyout = (item: IWaterfallItem) => {
-    this.setQueryParams({
-      flyoutDetailTab: undefined,
-      waterfallItemId: String(item.id)
-    });
-  };
+const toggleFlyout = ({
+  item,
+  location
+}: {
+  item?: IWaterfallItem;
+  location: Location;
+}) => {
+  history.replace({
+    ...location,
+    search: fromQuery({
+      ...toQuery(location.search),
+      ...{
+        flyoutDetailTab: undefined,
+        waterfallItemId: item ? String(item.id) : undefined
```

Comment:
I think `item.id` is always a string, so this might not be necessary anymore (not 100% so please double check).

You should therefore be able to do:
```diff
interface IWaterfallItemBase {
-  id: string | number;
+  id: string;
```